cedar-policy-core/src/parser/err.rs

```diff
@@ -47,9 +54,288 @@ pub enum ParseError {
+    InvalidPattern(String),
+    /// Returned when an unexpected node is in the policy scope clause
+    #[error("expected a {expected}, found a `{got}` statement")]
+    WrongNode {
+        /// What the expected AST node kind was
+        expected: &'static str,
+        /// What AST node was present in the policy source
+        got: String,
+    },
+    /// Returned when a policy contains ambiguous ordering of operators.
+    /// This can be resolved by using parenthesis to make order explicit
+    #[error("multiple relational operators (>, ==, in, etc.) without parentheses")]
+    AmbiguousOperators,
+    /// Returned when a policy uses the division operator (`/`), which is not supported
+    #[error("division is not supported")]
+    UnsupportedDivision,
+    /// Returned when a policy uses the remainder/modulo operator (`%`), which is not supported
+    #[error("remainder/modulo is not supported")]
+    UnsupportedModulo,
+    /// Returned when a policy attempts to multiply by a non-constant integer
+    #[error("multiplication must be by an integer literal")]
+    NonConstantMultiplication,
+    /// Returned when a policy contains an integer literal that is out of range
+    #[error(
+        "integer literal `{0}` is too large. Maximum allowed integer literal is `{}`",
+        i64::MAX
+    )]
+    IntegerLiteralTooLarge(u64),
+    /// Returned when a unary operator is chained more than 4 times in a row
+    #[error(
+        "too many occurrences of `{0}`. Cannot chain more the 4 applications of a unary operator"
+    )]
+    UnaryOppLimit(crate::ast::UnaryOp),
+    /// Returned when a variable is called as a function, which is not allowed.
+    /// Functions are not first class values in Cedar
+    #[error("variables cannot be called as functions. `{0}(...)` is not a valid function call")]
+    VariableCall(crate::ast::Var),
+    /// Returned when a policy attempts to call a method on a value that has no methods
+    #[error("`{0}` does not have any methods")]
+    NoMethods(crate::ast::Name),
+    /// Returned when a policy attempts to call a function that does not exist
+    #[error("`{0}` is not a function")]
+    NotAFunction(crate::ast::Name),
+    /// Returned when a policy attempts to lookup an arbitrary entity
+    #[error("arbitrary entity lookups are not supported")]
+    ArbitraryEntityLookup,
+    /// Returned when an expression is the target of a function call.
+    /// Functions are not first class values in Cedar
+    #[error("function calls in cedar must be of the form: `<name>(arg1, arg2, ...)`")]
+    ExpressionCall,
+    /// Returned when a policy attempts to access the fields of a value with no fields
+    #[error("`{0}` has no fields or methods")]
+    InvalidAccess(crate::ast::Name),
+    /// Returned when a string-literal is expected but not found
+    #[error("`{0}` is not a string literal")]
+    NonStringLiteral(cst::Primary),
```

Comment:
This seems to be a superset of other errors like `NonStringIndex` and `InvalidPattern`. Can it be more specific?